test/blackbox-tests/test-cases/install-with-var/run.t

```diff
@@ -24,3 +24,102 @@
     "_build/install/default/man/man-page-a.default.1" {"man-page-a.default.1"}
     "_build/install/default/man/man1/man-page-b.default.1"
     "_build/install/default/man/man3/another-man-page.3"
+
+Some variables are restricted in [dst] of [bin] section because evaluating
+them could cause a dependency cycle (also, most of them make no sense in [dst] anyway).
+
+  $ unset FOO
+
+  $ cat > dune <<EOF
+  >  (install
+  >   (section bin)
+  >   (files (foobar.txt as "%{env:FOO=foobar}/foo.txt"))
+  >  )
+  > EOF
+
+  $ dune build @install
+  File "dune", line 3, characters 27-42:
+  3 |   (files (foobar.txt as "%{env:FOO=foobar}/foo.txt"))
+                                 ^^^^^^^^^^^^^^^
+  Error: %{env:..} isn't allowed in this position
+  [1]
+
+This is not a problem outside of bin section:
+
+  $ cat > dune <<EOF
+  >  (install
+  >   (section man)
+  >   (files (foobar.txt as "%{env:FOO=foobar}/foo.txt"))
+  >  )
+  > EOF
+
+  $ dune build @install
+
+Extension of [src] can't use the restricted variables either because
+addition of .exe suffix to dst on Windows is conditional on the
+extension of [src]:
+
+  $ cat > dune <<EOF
+  >  (install
+  >   (section bin)
+  >   (files (%{env:FOO=foobar} as foo.txt))
+  >  )    
+  > EOF
+
+  $ dune build @install
+  File "dune", line 3, characters 12-27:
+  3 |   (files (%{env:FOO=foobar} as foo.txt))
+                  ^^^^^^^^^^^^^^^
+  Error: Because this file is installed in the 'bin' section, you
+  cannot use the variable %{env:..} in its basename.
+  [1]
+
+This is fine if the destination extension is already .exe:
+
+  $ cat > dune <<EOF
+  >  (install
+  >   (section bin)
+  >   (files (%{env:FOO=foobar}.txt as foo.exe))
+  >  )    
+  > EOF
+
+  $ dune build @install
+
+Or if the extension of source is clearly not .exe:
+
+  $ cat > dune <<EOF
+  >  (install
+  >   (section bin)
+  >   (files (%{env:FOO=foobar}.txt as foo.exe))
+  >  )    
+  > EOF
```

Comment:
The last two tests are identical, is that on purpose?

Reply:
No, of course not. Fixed.